tools/yaml-templates/ios-test.yml

```diff
@@ -147,11 +147,16 @@ steps:
 
       echo "ls -lR ${output}"
       ls -lR "${output}"
+
+      echo "Zipping logs directory..."
+      cd "$(agent.buildDirectory)/Logs/$(agent.JobName)"
+      zip -r "$(agent.JobName)_DebugLogs.zip" .
```

Comment:
>$(agent.JobName)_DebugLogs.zip [](http://example.com/codeflow?start=14&length=30)

Can this string be set to a YAML variable/constant so you don't have to duplicate it here and then again on line 159?

Perhaps you could do the same with the path? E.g.
```
variables:
  logsDirectory: "$(agent.buildDirectory)/Logs/$(agent.JobName)"
  debugLogsZip: "$(agent.JobName)_DebugLogs.zip"
...
      echo "Zipping logs directory..."
      cd "${{ variables.logsDirectory }}"
      zip -r "${{ variables.debugLogsZip }}" .
...
  - task: 1ES.PublishPipelineArtifact@1
    inputs:
      path: '${{ variables.logsDirectory }}/${{ variables.debugLogsZip }}'
```

...or something like that?

Reply:
We can have variables but that will be defined in `azure-pipelines.yml`. I feel like in that way it doesn't make too much sense to other yml files. You can check here that I made a change under iOS yml file and it cannot be parsed correctly. So maybe I'll keep it as is?

https://office.visualstudio.com/ISS/_build/results?buildId=33089221&view=results

Reply:
I didn't know that YAML imposes such a restriction on where variables can be defined. Seems weird. I guess my encouragement then would be "reuse as much as you can to avoid duplication within the limits of what YAML allowsl."

Reply:
YAML file restrictions are weird. It looks like you are writing some type of code but actually it doesn't. Even if you take look on their documentation, their examples show that you can add `variables` under the job. But only when you write something and give it a try, you know that `variables` can only be defined in the most upper level of yml file.